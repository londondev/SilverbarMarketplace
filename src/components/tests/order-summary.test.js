import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import  OrderSummary  from '../order-summary';

const defaultProps = {
  orderSummary: [{unitPrice:100, quantity:200}, {unitPrice:200, quantity:400}],
  tableHeader: 'Test header'
};

describe('Order Summary', () => {

  it('render without crashing', () => {
    shallow(<OrderSummary {...defaultProps} />);
  });
  it('should render orders', () => {
    const renderedItem = shallow(<OrderSummary {...defaultProps}  />);
    expect(renderedItem.find('tr')).toHaveLength(4);
  });
  it('should match the snapshot', () =>{
    const renderedItem = renderer
    .create(<OrderSummary {...defaultProps} />)
    .toJSON();
    expect(renderedItem).toMatchSnapshot();
  })
});