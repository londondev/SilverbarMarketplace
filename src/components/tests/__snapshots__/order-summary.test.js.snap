// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Summary should match the snapshot 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <th
          colSpan="2"
        >
          Order Summary(
          Test header
          )
        </th>
      </tr>
      <tr>
        <th>
          Unit Price
        </th>
        <th>
          Quantity
        </th>
      </tr>
      <tr>
        <td>
          £
          100
        </td>
        <td>
          200
        </td>
      </tr>
      <tr>
        <td>
          £
          200
        </td>
        <td>
          400
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
